operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -81,6 +93,18 @@ protected Integer currentScale(String namespace, String name) {
             }
         }
 
+        copyClusterCaCertGeneration(namespace, desired);
         return super.internalPatch(namespace, name, current, desired, cascading);
     }
+
+    private static ObjectMeta templateMetadata(Deployment resource) {
+        return resource.getSpec().getTemplate().getMetadata();
+    }
+
+    private void copyClusterCaCertGeneration(String namespace, Deployment desired) {
```

Comment:
Can't you factor out a method for dealing with these annotations that's common between deployment and statefulset?

Reply:
Yeah sure, it was my plan.